Some regtests fail on ARM (aarch64)

Hello, I meet a problem:cp2k@9.1  test failed on centos8_aarch64.
### Steps to reproduce the issue

```console
[root@localhost ~]git clone --recursive https://github.com/cp2k/cp2k.git cp2k
[root@localhost ~]cd cp2k
[root@localhost cp2k]#cd tools/toolchain
[root@localhost cp2k]#./install_cp2k_toolchain.sh --with-libxsmm=no  --with-libvori=no
[root@localhost cp2k]#cd ../..
[root@localhost cp2k]#cp tools/toolchain/install/arch/* arch/
[root@localhost cp2k]#source /tools/toolchain/install/setup
[root@localhost cp2k]#yum install liblsan.aarch64 gcc-toolset-10-liblsan-devel.aarch64
[root@localhost cp2k]#make -j 128 ARCH=local VERSION="ssmp sdbg"
[root@localhost cp2k]#make -j 128 ARCH=local VERSION="ssmp sdbg" test
……
    cc2_lri.inp                                                                    OK (  11.08 sec)
<<< /home/stage/root/spack-stage-cp2k-9.1-d2yemk6wqui237fwkyhdwezq6yl3jt5c/spack-src/regtesting/TEST-local-ssmp-2022-01-04_17-02-45/QS/regtest-kp-2 (286 of 286) done in 382.49 sec
------------------------------- Errors ---------------------------------
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
/home/stage/root/spack-stage-cp2k-9.1-d2yemk6wqui237fwkyhdwezq6yl3jt5c/spack-src/regtesting/TEST-local-ssmp-2022-01-04_17-02-45/QS/regtest-gw-cubic/G0W0_kpoints_from_Gamma.inp.out
Difference too large: 7.56e-02 > 1e-05, ref_value: 14.55, value: 13.45.
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
/home/stage/root/spack-stage-cp2k-9.1-d2yemk6wqui237fwkyhdwezq6yl3jt5c/spack-src/regtesting/TEST-local-ssmp-2022-01-04_17-02-45/QS/regtest-gw-cubic/G0W0_kpoints_from_Gamma_RI_regularization_1E-3.inp.out
Difference too large: 1.18e-01 > 1e-05, ref_value: 14.71, value: 12.97.
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
/home/stage/root/spack-stage-cp2k-9.1-d2yemk6wqui237fwkyhdwezq6yl3jt5c/spack-src/regtesting/TEST-local-ssmp-2022-01-04_17-02-45/QS/regtest-hfx-ri-2/H2O-hfx-stress-identity.inp.out
Difference too large: 4.68e-01 > 1e-08, ref_value: 8.19144120830E-01, value: 4.35583649823E-01.
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
/home/stage/root/spack-stage-cp2k-9.1-d2yemk6wqui237fwkyhdwezq6yl3jt5c/spack-src/regtesting/TEST-local-ssmp-2022-01-04_17-02-45/xTB/regtest-5/AdeThyvdW.inp.out


Timed out after 400.09734279010445 seconds.

------------------------------- Timings --------------------------------
Plot: name="timings", title="Timing Distribution", ylabel="time [s]"
PlotPoint: name="100th_percentile", plot="timings", label="100th %ile", y=400.10, yerr=0.0
PlotPoint: name="99th_percentile", plot="timings", label="99th %ile", y=103.18, yerr=0.0
PlotPoint: name="98th_percentile", plot="timings", label="98th %ile", y=75.22, yerr=0.0
PlotPoint: name="95th_percentile", plot="timings", label="95th %ile", y=44.12, yerr=0.0
PlotPoint: name="90th_percentile", plot="timings", label="90th %ile", y=25.62, yerr=0.0
PlotPoint: name="80th_percentile", plot="timings", label="80th %ile", y=12.04, yerr=0.0

------------------------------- Summary --------------------------------
Number of FAILED  tests 1
Number of WRONG   tests 3
Number of CORRECT tests 3609
Total number of   tests 3613

Summary: correct: 3609 / 3613; wrong: 3; failed: 1; 10min
Status: FAILED

*************************** Testing ended ******************************
make[3]: *** [/home/stage/root/spack-stage-cp2k-9.1-d2yemk6wqui237fwkyhdwezq6yl3jt5c/spack-src/Makefile:266: test] Error 4
make[2]: *** [/home/stage/root/spack-stage-cp2k-9.1-d2yemk6wqui237fwkyhdwezq6yl3jt5c/spack-src/Makefile:163: test] Error 2
make[1]: *** [/home/stage/root/spack-stage-cp2k-9.1-d2yemk6wqui237fwkyhdwezq6yl3jt5c/spack-src/Makefile:140: ssmp] Error 2
make[1]: *** Waiting for unfinished jobs....
```
Can you help me?This seems to have something to do with the tolerance.

--- a/src/dbt/tas/dbt_tas_types.F
+++ b/src/dbt/tas/dbt_tas_types.F
@@ -62,9 +62,9 @@ MODULE dbt_tas_types
    TYPE dbt_tas_mm_storage
       TYPE(dbt_tas_type), POINTER :: store_batched => NULL() ! intermediate replicated matrix
       TYPE(dbt_tas_type), POINTER :: store_batched_repl => NULL() ! intermediate replicated matrix
-      LOGICAL :: batched_out ! whether replicated matrix has been changed in mm and should be copied to actual matrix
-      LOGICAL :: batched_trans
-      REAL(dp) :: batched_beta
+      LOGICAL :: batched_out = .FALSE. ! whether replicated matrix has been changed in mm and should be copied to actual matrix
+      LOGICAL :: batched_trans = .FALSE.
+      REAL(dp) :: batched_beta = 1.0_dp
    END TYPE
 
    ! type for tall-and-skinny matrices

